### ISSUE TITLE
Extraction of subset of binary MITgcm data

### ISSUE THREAD

**Issue by SOFIADAL** [MITgcm/xmitgcm#297]
Hello,

I am a new MITgcm file user and I was wondering whether it is possible to extract specific longitude/latitude subregions from a binary file, the same way it is done in Netcdf files. I would like to also save the extract file into a new binary file. Is this possible with xmitgcm?

Thank you in advance for your help,

**Comment by rabernat** [comment]
Yes. If the MITgcm data uses a lat lon grid, xmitgcm.open_mdsdataset will return an [Xarray Dataset](https://xarray.pydata.org/en/stable/user-guide/data-structures.html#dataset). You can use that dataset to do label-based data selection. E.g.

```python
from xmitgcm import open_mdsdataset
ds = open_mdsdataset(<YOUR OPTIONS GO HERE>)
ds.sel(lon=slice(30, 40), lat=slice(50, 60))
```

**Comment by rabernat** [comment]
> I would like to also save the extracted file into a new binary file.

You can export your data to netCDF from xarray, `e.g. ds.to_netcdf`. 

It is possible write MDS data with xmitgcm as well, but this is not very well supported: https://xmitgcm.readthedocs.io/en/latest/demo_writing_binary_file.html

**Comment by SOFIADAL** [comment]
ok @rabernat thanks a lot for the quick answer. It is very helpful.
